git/working.go

```diff
@@ -20,8 +20,10 @@ type Config struct {
 	NotesRef    string
 	UserName    string
 	UserEmail   string
+	SigningKey  string
 	SetAuthor   bool
 	SkipMessage string
+	GPGHomeDir  string
```

Comment:
This only seems to be used for testing and it is passed on to various commands. 

I am not sure if it can constitute an attack vector, but at least the code would be cleaner by simply defining the `GNUPGHOME` variable in the tests.

Reply:
This means adding it to the allowed env var list - which does result in better protection - but is cleaner.

Reply:
How about just temporarily adding it `allowedEnvVars` during the test?

Reply:
I have been playing around with the idea (after the brief discussion we had in private) and I think `GNUPGHOME` should still be configurable from _outside_ the package.

Overwriting the var in a test in the `git` package is great but prevents packages using the machinery to sign commits as they have no control over `GNUPGHOME`.

This leaves us with two options; we either put it into the `Config` struct or we allow inheritance of the `GNUPGHOME` env variable. The latter being a lot cleaner and even helpful to our users as they now have control over where their imported keys are stored.

Reply:
I just found a 3rd reason: in combination with #1156 this would work outside a cluster.